EasyModbusSource/src/main/java/io/vantiq/extsrc/EasyModbusSource/EasyModbusMain.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2020 Vantiq, Inc.
+ *
+ * All rights reserved.
+ * 
+ * SPDX: MIT
+ */
+
+package io.vantiq.extsrc.EasyModbusSource;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The main class of this program. It connects to sources using the specified
+ * configuration. All sources must belong to the same VANTIQ Server & Namespace
+ * (i.e. same authtoken). Options are in a file where each option has its own
+ * line, and is in the form {@code property=value} with no quotes required for
+ * Strings. Options are:
+ * <ul>
+ * <li>{@code authToken}: Required. The authentication token to connect with.
+ * These can be obtained from the namespace admin.
+ * <li>{@code sources}: Required. A comma separated list of the sources to which
+ * you wish to connect. Whitespace around each name will be removed when read.
+ * <li>{@code targetServer}: Required. The Vantiq server hosting the sources.
+ * </ul>
+ */
+public class EasyModbusMain {
+
+    static final Logger log = LoggerFactory.getLogger(EasyModbusMain.class);
+    static List<EasyModbusCore> sources;
+
+    static String authToken;
+    static String targetVantiqServer;
+
+    // Exit Error Codes
+    static final int NO_AUTH_EXIT = 1;
+    static final int NO_SOURCE_EXIT = 2;
+    static final int NO_SERVER_EXIT = 3;
+
+    /**
+     * Connects to the Vantiq source and starts polling for data. Exits when all
+     * sources are done running.
+     * 
+     * @param args Should be either null or the first argument as a config file
+     */
+    public static void main(String[] args) {
+        Properties config;
+        if (args != null && args.length > 0) {
+            config = obtainServerConfig(args[0]);
+        } else {
+            config = obtainServerConfig("server.config");
+        }
+
+        sources = createSources(config);
+
+        startSources(sources);
+
+        // Can leave now because the threads created by the sources' WebSocket
+        // connections will keep the JVM alive
+    }
+
+    /**
+     * Starts every source, giving each up to 10 seconds to connect. Starts them in
+     * new threads so they don't block.
+     * 
+     * @param sources The list of sources which should be started.
+     */
+    private static void startSources(List<EasyModbusCore> sources) {
+        for (EasyModbusCore source : sources) {
+            // Starting in threads so they can all connect at once
+            new Thread(() -> {
+                source.start(10);
+            }).start();
+            ;
```

Comment:
not sure this is adding much :-). Please remove

Reply:
its the same pattern being used in all extension source . the ability to run multiple object might be important in other cases, but this is kind of  template that is being used . do you still wants me to remove it ? ( no problem to do that...)

Reply:
The part to remove is line 83 -- the line with just a ';' on it.  We don't need the null statement.